### DISCUSSION TITLE
Inline editing with select2 or similar

### DISCUSSION THREAD

**Discussion by dgillier** [rappasoft/laravel-livewire-tables#1181] [Feature Requests]
Hello !

I'm wondering if someone has tried to do inline editing inside the table.

i'm looking to have a select2 or similar component displaying user rol.
If current user is having the proper role, he can inline editing the select2 content.

Any suggestions ?
Thanks, Denis

**Comment by lrljoe** [comment]
So I looked at this a while back.

The only way that made sense to me was a dynamic component that load when you click into the field, as otherwise having a hundred (depending on pagination count) components turns into a memory nightmare.

You could look at the WireUI Modal to get an example of how to implement this smoothly.

It's something that'll almost certainly be in v3, but that's a long way off!

**Reply by lrljoe** [reply]
So just found the right branch for what I was testing with.

The way I've done it is have a component sitting in the background listening.  

When I click on a field, it emits the primary key to the component.

It then replaces the row with its view, let's you edit the fields, and when you click save, it saves the model and refreshes the table.

It does mean having a trait or component for each table as otherwise you'll run into issues with it not knowing how to present each column (e.g. select, text entry, toggle).

The way I tackled this was each model has an array that tells the component how to present each field, where it's not obvious from the cast array,, but that was definitely a headache to implement on a larger project.

I reverted to just using WireUI modals as they're easy to work with.
